augur/tasks/github/util/github_data_access.py

```diff
@@ -111,10 +111,26 @@ def make_request(self, url, method="GET", timeout=100):
 
             if response.status_code in [403, 429]:
                 self.expired_keys_for_request.append(self.key)
+                self.logger.warning(f"Github rate limit exceeded. Key: {self.key[-5:]}. Response: {response.text}")
                 raise RatelimitException(response, self.expired_keys_for_request[-5:])
 
+            # There are cases with PR files, PR commits, and messages where the parent object is removed after 
+            # It is collected, leading the the associated URL for those objects to return a 404. 
+            # This is not an issue that is really an Exception. It is more of a nominal signal. 
+            
             if response.status_code == 404:
-                raise UrlNotFoundException(f"Could not find {url}")
+                parsed = urlparse(url)
+                path = parsed.path.lower()
+                query = parsed.query.lower()
+
+                if any(k in path for k in ["commits", "files", "comments"]) or \
+                any(k in query for k in ["commits", "files", "comments"]):
+                    self.logger.warning(
+                        f"Github response with 404 for PR files, PR commits or messages. "
+                        f"This is a data anomaly in the platform API, not an error. URL: {url}. Response: {response.text}"
+                    )
+                else:
+                    raise UrlNotFoundException(f"Could not find {url}")
```

Comment:
This logic should live in the pr files, pr commits, and pt comments collection files. 

It's not the responsibility of GitHub data access to understand how to handle responses from the GitHub api. Its responsibilities are to communicate the result of data retrieval (success by not throwing an exception or failure by throwing an exception), handle authentication and ensure fault tolerance from rate limits.

Essentially each of those needs to wrap the call to GitHub data access in a try catch and the specifically catch the UrlNotFoundException. Then the the catch you can do whatever logic that's needed for the specific data that's being collected.

It looks like it is already handled for files and messages, but not for commits

Reply:
Are you saying when we request all messages for a repo and it doesn't have any it will return a 404? 

If so we should catch the UrlNotFoundException in the call to /<owner>/<repo>/messages